### ISSUE TITLE
Newly created scene does not work

### ISSUE THREAD

**Issue by ruzzbug1** [icanos/hassio-plejd#125]
I have a newly created scene (ID16) that doesn't work to trig from HA. The old ones work perfectly. Tried creating another scene (ID17) but that wouldn't work either.
FYI, also had problem getting renamed units (dim-01 and ctr-01) into HA, but that was resolved with a reboot of my rpi4.

> startWriteQueue()
plejd-ble: sending 8 byte(s) of data to Plejd
plejd-ble: sending 8 byte(s) of data to Plejd
plejd-ble: sending 8 byte(s) of data to Plejd
plejd-ble: sending 8 byte(s) of data to Plejd
plejd-ble: sending 8 byte(s) of data to Plejd
triggering scene with ID 16
plejd-ble: sending 6 byte(s) of data to Plejd
triggering scene with ID 16
plejd-ble: sending 6 byte(s) of data to Plejd
triggering scene with ID 16
plejd-ble: sending 6 byte(s) of data to Plejd
triggering scene with ID 16
plejd-ble: sending 6 byte(s) of data to Plejd
triggering scene with ID 16
plejd-ble: sending 6 byte(s) of data to Plejd
triggering scene with ID 16
plejd-ble: sending 6 byte(s) of data to Plejd
triggering scene with ID 16
plejd-ble: sending 6 byte(s) of data to Plejd
triggering scene with ID 16
plejd-ble: sending 6 byte(s) of data to Plejd
triggering scene with ID 16
plejd-ble: sending 6 byte(s) of data to Plejd
triggering scene with ID 16
triggering scene with ID 16
plejd-ble: sending 6 byte(s) of data to Plejd
plejd-ble: sending 6 byte(s) of data to Plejd
triggering scene with ID 16
plejd-ble: sending 6 byte(s) of data to Plejd
plejd-ble: sending 6 byte(s) of data to Plejd
plejd-ble: sending 6 byte(s) of data to Plejd
plejd-ble: sending 6 byte(s) of data to Plejd
plejd-ble: sending 8 byte(s) of data to Plejd
triggering scene with ID 16
plejd-ble: sending 6 byte(s) of data to Plejd
triggering scene with ID 16
plejd-ble: sending 6 byte(s) of data to Plejd
triggering scene with ID 5
plejd-ble: sending 6 byte(s) of data to Plejd
plejd-ble: sending 8 byte(s) of data to Plejd
plejd-ble: sending 8 byte(s) of data to Plejd
plejd-ble: sending 8 byte(s) of data to Plejd
plejd-ble: sending 8 byte(s) of data to Plejd
plejd-ble: sending 8 byte(s) of data to Plejd
error: plejd ping failed
plejd-ble: ping failed, reconnecting.
init()
error: writing to plejd: DBusError: Not connected
plejd-ble: ping failed, reconnecting.
init()
(node:301) UnhandledPromiseRejectionWarning: DBusError: Does Not Exist
    at _methodReturnHandlers.<computed> (/plejd/node_modules/dbus-next/lib/bus.js:339:27)
    at handleMessage (/plejd/node_modules/dbus-next/lib/bus.js:98:11)
    at EventEmitter.<anonymous> (/plejd/node_modules/dbus-next/lib/bus.js:147:9)
    at EventEmitter.emit (events.js:315:20)
    at /plejd/node_modules/dbus-next/lib/connection.js:112:14
    at Socket.<anonymous> (/plejd/node_modules/dbus-next/lib/message.js:63:9)
    at Socket.emit (events.js:315:20)
    at emitReadable_ (_stream_readable.js:555:12)
    at processTicksAndRejections (internal/process/task_queues.js:83:21)
(node:301) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:301) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
plejd-ble: connecting to /org/bluez/hci0/dev_E2_57_0D_A6_01_0A
plejd-ble: warning: unable to connect, will retry. DBusError: Software caused connection abort
plejd-ble: connecting to /org/bluez/hci0/dev_E2_EB_49_51_3A_D2
onDeviceConnected()
trying 10 characteristics
authenticate()
startPing()
startWriteQueue()
triggering scene with ID 17
plejd-ble: sending 8 byte(s) of data to Plejd
triggering scene with ID 17
plejd-ble: sending 8 byte(s) of data to Plejd
triggering scene with ID 17
triggering scene with ID 17
plejd-ble: sending 8 byte(s) of data to Plejd
error: writing to plejd: DBusError: Not connected
plejd-ble: ping failed, reconnecting.
init()
plejd-ble: write failed DBusError: Not connected
init()
plejd-ble: error: failed to start discovery. Make sure no other add-on is currently scanning.
onDeviceConnected()
plejd-ble: error: unable to enumerate characteristics.
plejd-ble: were unable to connect, will retry connection in 10 seconds.
triggering scene with ID 17
triggering scene with ID 17
init()
plejd-ble: connecting to /org/bluez/hci0/dev_C9_9F_12_9E_83_CC
triggering scene with ID 17
onDeviceConnected()
trying 10 characteristics
authenticate()
startPing()
startWriteQueue()
plejd-ble: sending 8 byte(s) of data to Plejd
plejd-ble: sending 8 byte(s) of data to Plejd
plejd-ble: sending 8 byte(s) of data to Plejd

**Comment by SweVictor** [comment]
I've been looking a lot at how things work lately and going through logs. I sometimes get a lot of UnhandledPromiseRejectionWarnings as well, then `hassio-plejd` goes into reconnecting mode. I'm guessing that it loses connection to the Plejd device, but I think the error handling around this could/should be improved.

Some logs from my system recently (happenes to me sometimes when triggering lots of changes at the same time, basically transitioning ALL lights, possibly due to bad BLE connection). Logs often overflow in addon UI so its hard to see before first log row. Is there any way to see/copy entire log? After reproducing a few times it seems like log row before/after errors are not necessarily helpful. Seems like it could be any comm-related log (send or receive event).

I'm quite sure **messages are lost** when this (and other reconnects) happens, maybe it would be a good idea to re-add messages in queue when errors occur?

```
(node:300) UnhandledPromiseRejectionWarning: DBusError: Does Not Exist
    at _methodReturnHandlers.<computed> (/plejd/node_modules/dbus-next/lib/bus.js:339:27)
    at handleMessage (/plejd/node_modules/dbus-next/lib/bus.js:98:11)
    at EventEmitter.<anonymous> (/plejd/node_modules/dbus-next/lib/bus.js:147:9)
    at EventEmitter.emit (events.js:315:20)
    at /plejd/node_modules/dbus-next/lib/connection.js:112:14
    at Socket.<anonymous> (/plejd/node_modules/dbus-next/lib/message.js:63:9)
    at Socket.emit (events.js:315:20)
    at emitReadable_ (_stream_readable.js:555:12)
    at processTicksAndRejections (internal/process/task_queues.js:83:21)
(node:300) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 15)
(node:300) UnhandledPromiseRejectionWarning: DBusError: Does Not Exist
    at _methodReturnHandlers.<computed> (/plejd/node_modules/dbus-next/lib/bus.js:339:27)
    at handleMessage (/plejd/node_modules/dbus-next/lib/bus.js:98:11)
    at EventEmitter.<anonymous> (/plejd/node_modules/dbus-next/lib/bus.js:147:9)
    at EventEmitter.emit (events.js:315:20)
    at /plejd/node_modules/dbus-next/lib/connection.js:112:14
    at Socket.<anonymous> (/plejd/node_modules/dbus-next/lib/message.js:63:9)
    at Socket.emit (events.js:315:20)
    at emitReadable_ (_stream_readable.js:555:12)
    at processTicksAndRejections (internal/process/task_queues.js:83:21)
(node:300) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 16)
(node:300) UnhandledPromiseRejectionWarning: DBusError: Does Not Exist
    at _methodReturnHandlers.<computed> (/plejd/node_modules/dbus-next/lib/bus.js:339:27)
    at handleMessage (/plejd/node_modules/dbus-next/lib/bus.js:98:11)
    at EventEmitter.<anonymous> (/plejd/node_modules/dbus-next/lib/bus.js:147:9)
    at EventEmitter.emit (events.js:315:20)
    at /plejd/node_modules/dbus-next/lib/connection.js:112:14
    at Socket.<anonymous> (/plejd/node_modules/dbus-next/lib/message.js:63:9)
    at Socket.emit (events.js:315:20)
    at emitReadable_ (_stream_readable.js:555:12)
    at processTicksAndRejections (internal/process/task_queues.js:83:21)
(node:300) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 17)
(node:300) UnhandledPromiseRejectionWarning: DBusError: Does Not Exist
    at _methodReturnHandlers.<computed> (/plejd/node_modules/dbus-next/lib/bus.js:339:27)
    at handleMessage (/plejd/node_modules/dbus-next/lib/bus.js:98:11)
    at EventEmitter.<anonymous> (/plejd/node_modules/dbus-next/lib/bus.js:147:9)
    at EventEmitter.emit (events.js:315:20)
    at /plejd/node_modules/dbus-next/lib/connection.js:112:14
    at Socket.<anonymous> (/plejd/node_modules/dbus-next/lib/message.js:63:9)
    at Socket.emit (events.js:315:20)
    at emitReadable_ (_stream_readable.js:555:12)
    at processTicksAndRejections (internal/process/task_queues.js:83:21)
(node:300) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 18)
(node:300) UnhandledPromiseRejectionWarning: DBusError: Does Not Exist
    at _methodReturnHandlers.<computed> (/plejd/node_modules/dbus-next/lib/bus.js:339:27)
    at handleMessage (/plejd/node_modules/dbus-next/lib/bus.js:98:11)
    at EventEmitter.<anonymous> (/plejd/node_modules/dbus-next/lib/bus.js:147:9)
    at EventEmitter.emit (events.js:315:20)
    at /plejd/node_modules/dbus-next/lib/connection.js:112:14
    at Socket.<anonymous> (/plejd/node_modules/dbus-next/lib/message.js:63:9)
    at Socket.emit (events.js:315:20)
    at emitReadable_ (_stream_readable.js:555:12)
    at processTicksAndRejections (internal/process/task_queues.js:83:21)
(node:300) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 19)
(node:300) UnhandledPromiseRejectionWarning: DBusError: Does Not Exist
    at _methodReturnHandlers.<computed> (/plejd/node_modules/dbus-next/lib/bus.js:339:27)
    at handleMessage (/plejd/node_modules/dbus-next/lib/bus.js:98:11)
    at EventEmitter.<anonymous> (/plejd/node_modules/dbus-next/lib/bus.js:147:9)
    at EventEmitter.emit (events.js:315:20)
    at /plejd/node_modules/dbus-next/lib/connection.js:112:14
    at Socket.<anonymous> (/plejd/node_modules/dbus-next/lib/message.js:63:9)
    at Socket.emit (events.js:315:20)
    at emitReadable_ (_stream_readable.js:555:12)
    at processTicksAndRejections (internal/process/task_queues.js:83:21)
(node:300) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 20)
plejd-ble reconnected and retrying to write
...
```

**Comment by JohnPhoto** [comment]
I got these errors but managed to resolvera them in #137.. the DBus error is because you are getting multiple re-inits at the same time and competing with yourself of dbus ”connections”, racecondition-style.

**Comment by SweVictor** [comment]
@ruzzbug1 : Has the problem been solved after #137 or possibly after the last changes merged? If not, could you set logLevel to verbose and get back with what happens?

**Comment by SweVictor** [comment]
Should be fixed either in #137 or in code merged to [develop branch](https://github.com/icanos/hassio-plejd/tree/develop), planned for 0.6.0 release. Closing
